src/serde/error.rs

```diff
@@ -0,0 +1,64 @@
+#![cfg(feature = "serde")]
```

Comment:
Why is this needed?

Reply:
My `cfg` were all over the place, mostly because I refactored once or twice as I went. Cleaned up now, here and elsewhere.